### PR TITLE
[#116373301] Add "projects" engine

### PR DESCRIPTION
**Pull request by drench**
This adds engine boilerplate, a Project model with migration, Facility association, and validation specs.

I wanted to get a foundation out for review with changes to add routes, controller(s), view and Ability hooks coming in the next PR.

### CODE REVIEW

File: vendor/engines/projects/app/models/projects/project.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Projects
+
+  class Project < ActiveRecord::Base
+
+    belongs_to :facility, foreign_key: :facility_id
+    attr_accessible :description, :name
```
Comment (jhanggi) [inline vendor/engines/projects/app/models/projects/project.rb]: We have strong_params gem installed. Can we use that instead? Add `include ActiveModel::ForbiddenAttributesProtection` to the model. One less model we have to deal with when we upgrade rails.

File: vendor/engines/projects/spec/models/projects/project_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+require "rails_helper"
+
+RSpec.describe Projects::Project, type: :model do
+  subject(:project) { create(:project) }
```
Comment (jhanggi) [inline vendor/engines/projects/spec/models/projects/project_spec.rb]: Does this need to be `create` or could it be `build` or `build_stubbed`?

**Comment by jhanggi** [comment]
Just some minor comments; nothing major.

**Comment by drench** [comment]
Updated

File: vendor/engines/projects/app/models/projects/project.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module Projects
+
+  class Project < ActiveRecord::Base
+
+    include ActiveModel::ForbiddenAttributesProtection
+
+    belongs_to :facility, foreign_key: :facility_id
+    attr_accessible :description, :name
```
Comment (jhanggi) [inline vendor/engines/projects/app/models/projects/project.rb]: With `ActiveModel::ForbiddenAttributesProtection`, you don't need `attr_accessible` anymore.

File: vendor/engines/projects/app/models/projects/facility_extension.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  module FacilityExtension
+
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :projects,
+               class_name: "Projects::Project",
+               foreign_key: :facility_id,
+               inverse_of: :facility,
+               dependent: :destroy # Though Facilities cannot be destroyed
```
Comment (jhanggi) [inline vendor/engines/projects/app/models/projects/facility_extension.rb]: If facilities can't be destroyed, why do we need the `dependent: :destroy`?

Comment (drench) [inline vendor/engines/projects/app/models/projects/facility_extension.rb]: Facilities can't be destroyed yet, but if that ever changes, we'll be ready. We have similar `dependent: :destroy` relations for `order_imports` and `user_roles`.

**Comment by jhanggi** [comment]
👍

**Pull request merged**